text/0249-v2-experiments.md

```diff
@@ -408,3 +484,240 @@ Disadvantages:
+
+**Maven**
+
+No need to change the sources when using a stable library with an
+experimental app, as the namespaces and class names will be the same.
+
+Using a stable library in an experimental app can be done in two ways:
+
+* The library uses the `provided` dependency flavor (similar to `peerDependencies` in NPM).
+  Requires the app to bring the right version of CDK.
+* The app that wants to use the *experimental* build uses dependency exclusions
+  to remove the dependency on the *stable* build (looks like this is necessary for
+  every stable library individually)
+
+https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html#
+
+**pip**
+
+No need to change the sources when using a stable library with an
+experimental app, as the namespaces and class names will be the same.
+
+`pip` transitively fetches from the library's `setup.py`, and there's no way
+to override that.
+
+Our only option around that is to have stable libraries not declare any CDK
+dependency in `setup.py`, but then we also lose the ability for libraries to
+specify their dependency version, which is not really acceptable.
+
+Effectively, this seems to be a no-go.
+
+**.NET Core/NuGet**
+
+No need to change the sources when using a stable library with an
+experimental app, as the namespaces and class names will be the same.
+
+...pending responses from .NET SDK team...
+
+**Go**
+
+???
+
+## Prerelease tags
+
+We would vend the experimental build under a prerelease tag, for example:
+
+- `aws-cdk-lib@1.60.0`
+- `aws-cdk-lib@1.60.0-experimental`
+
+We could get rid of some of the ordering and warning problems by making sure the
+experimental version semver-orders *after* the stable version:
+
+- `aws-cdk-lib@1.60.100-experimental`
+- `aws-cdk-lib@1.61.0-experimental`
+
+But that is most likely just going to break people's heads so let's not even try.
+
+| .     | Stable apps | Experimental app uses Stable lib | Without complaints | Lib advertises Experimental |
+|-------|-------------|----------------------------------|--------------------|-----------------------------|
+| NPM   | yes         | yes                              | no                 | no                          |
+| Maven | no          | with excluded dependencies       | yes                | yes                         |
+| pip   | yes         | yes                              | no                 | yes                         |
+| NuGet | ?           | ?                                | ?                  | ?                           |
+| Go    | ?           | ?                                | ?                  | ?                           |
+
+**NPM**
+
+Stable ranges: yes, `^1.60.0` does not include `1.61.1-experimental`.
+
+Experimental app uses stable lib: can be done, the lib only uses a peerDependency. `npm ls` will complain about
+a missing peerDependency though.
+
+**Maven**
+
+Stable ranges: no, `[1.60.0,)` *will* include `1.61.1-experimental`.
```

Comment:
Again, I would question whether using a range for Maven is really interesting here.

(Also, is this for the library? Or app? It's really important to be precise here).